src/constants.ts

```diff
@@ -21,17 +21,41 @@ export const RELEASE_GROUP_REGEX =
 	/(?<=-)(?:\W|\b)(?!(?:\d{3,4}[ip]))(?!\d+\b)(?:\W|\b)(?<group>[\w .]+?)(?:\[.+\])?(?:\))?(?:\s\[.+\])?$/i;
 export const RESOLUTION_REGEX = /\b(?<res>\d{3,4}[pix](?:\d{3,4}[pi]?)?)\b/i;
 export const RES_STRICT_REGEX = /(?<res>(?:2160|1080|720)[pi])/;
-
+export const YEAR_REGEX = /(?<year>(?:19|20)\d{2})/;
 export const REPACK_PROPER_REGEX =
 	/(?:\b(?<type>(?:REPACK|PROPER|\d\v\d)\d?))\b/i;
-
 export const ARR_PROPER_REGEX = /(?:\b(?<arrtype>(?:Proper|v\d)))\b/;
-
 export const SCENE_TITLE_REGEX = /^(?:\w{0,5}-)?(?<title>.*)/i;
-
 export const ARR_DIR_REGEX =
 	/^(?!.*(?:(\d{3,4}[ipx])|([xh]26[4-6])|(?:(he)|a)vc))[\p{L}\s:\w'’!().,&–+-]+(?:\(\d{4}\))?\s?(?:\{(?:tm|tv|im)db(?:id)?-\w+\})?$/iu;
 export const SONARR_SUBFOLDERS_REGEX = /^(?:S(?:eason )?(?<season>\d{1,4}))$/i;
+export const NON_UNICODE_LETTER_NUM_REGEX = /[^\p{L}\p{N}]+/giu;
+
+// Needs to be hanlded through helper functions since there is varitions
+const SOURCE_REGEXES = {
+	AMZN: /\b(amzn|amazon(hd)?)\b/i,
+	DSNP: /\b(dsnp|dsny|disney)\b/i,
+	NF: /\b(nf|netflix(u?hd)?)\b/i,
+	HULU: /\b(hulu)\b/i,
+	ATVP: /\b(atvp|aptv)\b/i,
+	HBO: /\b(hbo)(?![ ._-]max)\b|\b(hmax|hbom|hbo[ ._-]max)\b|\b((?<!hbo[ ._-])max)\b/i,
+	PCOK: /\b(pcok|peacock)\b/i,
```

Comment:
same as above could potentially apply to peacock or max - dictionary words commonly found in episode or series titles...

Reply:
It actually doesn't matter for our use case too much. I tried to order the safer ones first which should catch most. Otherwise the chance of it happening is low. Even if it does, in safe mode it won't matter since they will be both be wrong but equal. And it's unlikely for something to get renamed that would remove one of these words for risky/partial. Same idea extends to ensemble.

We could also remove the problematic ones since which might be the best case. Unlikely the same release group gets renamed their source renamed between trackers.